### PR TITLE
feat(location)!: `nearbyGPSCoordinate` returns number tuple

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
Discussion reference #1058.

Changes the return type of `address.nearbyGPSCoordinate()` from a string tuple to a number tuple.

This is a breaking change and should not be merged before the next major release.

### CODE REVIEW

File: test/address.spec.ts [hunk 1]
```diff
@@ -487,42 +487,39 @@ describe('address', () => {
...
-                  latitude1,
-                  longitude1,
-                  latitude2,
-                  longitude2,
-                  isMetric
-                );
-                expect(actualDistance).toBeLessThanOrEqual(radius);
-              }
-            );
+            it(`should return random gps coordinate within a distance of another one (${JSON.stringify(
```
Comment (xDivisionByZerox) [inline test/address.spec.ts:490]: I removed the rerun of the test (`.each(times(5))`) because it doesn't bring any value other than extending the test time and bumping the number of test cases, IMO. 

If this was to catch some edge cases, the testcases designed should be changed instead of run the same test multiple times.

File: src/modules/address/index.ts [hunk 2]
```diff
@@ -439,19 +439,19 @@ export class Address {
...
   nearbyGPSCoordinate(
     coordinate?: [latitude: number, longitude: number],
     radius: number = 10,
     isMetric: boolean = false
-  ): [latitude: string, longitude: string] {
+  ): [latitude: number, longitude: number] {
     // If there is no coordinate, the best we can do is return a random GPS coordinate.
     if (coordinate === undefined) {
-      return [this.latitude(), this.longitude()];
+      return [parseFloat(this.latitude()), parseFloat(this.longitude())];
```
Comment (Shinigami92) [inline src/modules/address/index.ts:454]: Do we also want to change return type of `latitude` and `longitude`?

Comment (xDivisionByZerox) [reply src/modules/address/index.ts:454]: I wanted to provide for them as well. But since they are separate methods from this one I wanted to separate them and make this one depend on the other PR (or something like this).

IMO returning strings doesn't make sense for this method, but I think we are on the same side for this.

Comment (ST-DDT) [reply src/modules/address/index.ts:454]: You can remove the Todo for string inputs too.

**Review by ST-DDT — COMMENTED** [review]
The description text should include a hint that the user might want to use toFixed on the result and some reference how much another digit is in precision. E.g. 1 degree = 100 km. 1km = 0.01degree/ToFixed(2). We dont have to use precises values but some rough estimates. I guessed these values so you should probably verify them first.

File: src/modules/address/index.ts (hunk 2 above)
Comment (xDivisionByZerox) [reply src/modules/address/index.ts:454]: I wanted to leave them in since this PR technically doesn't fix what the todo states. But if you say so I will gladly remove it.

**Comment by xDivisionByZerox** [comment]
> The description text should include a hint that the user might want to use toFixed on the result and some reference how much another digit is in precision. E.g. 1 degree = 100 km. 1km = 0.01degree/ToFixed(2). We dont have to use precises values but some rough estimates. I guessed these values so you should probably verify them first.

Don't know about this. That wasn't the case before either. IMO, this is on the user side, since our responsibility is the (fake) data generation not being a wiki on how specific data should be handled. If someone requests GPS coordinates I would expect them to know how to work with them.

File: src/modules/address/index.ts [hunk 3]
```diff
@@ -491,7 +490,7 @@ export class Address {
     // Box longitude [-180°, 180°]
     newCoordinate[1] = (((newCoordinate[1] % 360) + 540) % 360) - 180;
 
-    return [newCoordinate[0].toFixed(4), newCoordinate[1].toFixed(4)];
+    return [newCoordinate[0], newCoordinate[1]];
```
Comment (ST-DDT) [inline src/modules/address/index.ts:493]: In the longitude and latitude PR you return the values with a specific presion.
IMO we should do the same here.

Comment (xDivisionByZerox) [reply src/modules/address/index.ts:493]: So you mean adding a parameter? Because currently, the user has no possibility to provide precision. I'm just working with what already exists here. If that is what you mean, I would want to provide an additional PR since this would be a feature (not like currently refactor) and is more comprehensible on the autogenerated changelog.

Otherwise, please elaborate on what you mean.

Comment (ST-DDT) [reply src/modules/address/index.ts:493]: Its okay to open a separate PR for it.
IMO the values should have the same precision regardless of whether the are genrated randomly (4) or nearby an existing location (10?).

File: src/modules/address/index.ts [hunk 4]
```diff
@@ -439,19 +439,18 @@ export class Address {
    * @param isMetric If `true` assume the radius to be in kilometers. If `false` for miles. Defaults to `false`.
    *
    * @example
-   * faker.address.nearbyGPSCoordinate() // [ '33.8475', '-170.5953' ]
-   * faker.address.nearbyGPSCoordinate([33, -170]) // [ '33.0165', '-170.0636' ]
-   * faker.address.nearbyGPSCoordinate([33, -170], 1000, true) // [ '37.9163', '-179.2408' ]
+   * faker.address.nearbyGPSCoordinate() // [ 33.8475, -170.5953 ]
+   * faker.address.nearbyGPSCoordinate([33, -170]) // [ 33.0165, -170.0636 ]
+   * faker.address.nearbyGPSCoordinate([33, -170], 1000, true) // [ 37.9163, -179.2408 ]
```
Comment (Shinigami92) [inline src/modules/address/index.ts:444]: Do we need to update these examples due to the output numbers could be now much more precise? (please see also https://github.com/faker-js/faker/pull/1064#discussion_r895224981)

Comment (xDivisionByZerox) [reply src/modules/address/index.ts:444]: I mean I did the bare minimum to achieve the signature change. But you are somewhat right. I'll provide some more diverse examples

**Review by ST-DDT — DISMISSED** [review]

**Comment by ST-DDT** [comment]
Considering the other PR, should we rather introduce a `format` parameter?

**Comment by xDivisionByZerox** [comment]
> Considering the other PR, should we rather introduce a `format` parameter?

It doesn't make sense to me that coordinates should be strings. We simply allow the input of stringified numbers due to the nature of JS. At least that's what I like to think.

**Comment by ST-DDT** [comment]
Cross referencing related PRs: 
- https://github.com/faker-js/faker/pull/1066
- https://github.com/faker-js/faker/pull/1064
- https://github.com/faker-js/faker/pull/1061

File: src/modules/address/index.ts (hunk 4 above)
Comment (ST-DDT) [reply src/modules/address/index.ts:388]: Please also address this.

File: src/modules/address/index.ts (hunk 2 above)
Comment (Shinigami92) [reply src/modules/address/index.ts:399]: @xDivisionByZerox Please extract it to a ticket then, so e don't miss it in v8.0

**Comment by Shinigami92** [comment]
I eagerly put the PR title to scope `location` as we will solve 
- #1344

in v8.0 and this will then be grouped together

File: src/modules/address/index.ts (hunk 2 above)
Comment (Shinigami92) [reply src/modules/address/index.ts:399]: Nevermind :slightly_smiling_face: 

- #1064

File: src/modules/address/index.ts (hunk 4 above)
Comment (Shinigami92) [reply src/modules/address/index.ts:388]: The default precision of lat and long functions is `4`!, so actually this is correct :grimacing:

Comment (Shinigami92) [reply src/modules/address/index.ts:388]: Or ... maybe we just should add a `precision` option? 🙃
The lat and long function also do have this option. :thinking:

Comment (xDivisionByZerox) [reply src/modules/address/index.ts:388]: This is not in the scope of this PR...

Was already suggested by @ST-DDT [here](https://github.com/faker-js/faker/pull/1061#discussion_r895200965)

Comment (Shinigami92) [reply src/modules/address/index.ts:388]: > This is not in the scope of this PR...

Yeah yes yes I know :sweat_smile: 
I just speak out loud

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]
